chain/rosetta-rpc/src/lib.rs

```diff
@@ -185,6 +198,73 @@ async fn network_options(
     }))
 }
 
+/// Get a block with `block_id`.
+/// Returns `Ok(Some(_))` if the block exists and is final.
+/// Returns `Ok(None)` if the block does not exist or is not final.
+async fn get_block_if_final(
+    block_id: &near_primitives::types::BlockReference,
+    view_client_addr: &Addr<ViewClientActor>,
+) -> Result<Option<near_primitives::views::BlockView>, models::Error> {
+    let final_block = match view_client_addr
+        .send(near_client::GetBlock(near_primitives::types::BlockReference::Finality(
+            near_primitives::types::Finality::Final,
+        )))
+        .await?
+    {
+        Ok(block) => block,
+        Err(_) => {
+            return Err(errors::ErrorKind::InternalError("final block not found".to_string()).into())
+        }
+    };
+    let is_query_by_height = match block_id {
+        near_primitives::types::BlockReference::Finality(
+            near_primitives::types::Finality::Final,
+        ) => return Ok(Some(final_block)),
+        near_primitives::types::BlockReference::BlockId(
+            near_primitives::types::BlockId::Height(height),
+        ) => {
+            if height > &final_block.header.height {
+                return Ok(None);
+            }
+            true
+        }
+        _ => false,
+    };
+    let block = match view_client_addr.send(near_client::GetBlock(block_id.clone())).await? {
+        Ok(block) => block,
+        Err(_) => return Ok(None),
+    };
+    // if block height is larger than the last final block height, then the block is not final
+    if block.header.height > final_block.header.height {
+        return Ok(None);
+    }
+    // check that this block is on the canonical chain
+    if is_query_by_height {
+        Ok(Some(block))
```

Comment:
I’d convert this to a `return Ok(Some(block));` so that the else part can be dedented.